core/arch/arm/plat-ti/main.c

```diff
@@ -130,12 +93,16 @@ struct plat_nsec_ctx {
 
 void init_sec_mon(unsigned long nsec_entry)
 {
-	struct plat_nsec_ctx *plat_ctx = (struct plat_nsec_ctx *)nsec_entry;
+	struct plat_nsec_ctx *plat_ctx;
 	struct sm_nsec_ctx *nsec_ctx;
 
+	plat_ctx = phys_to_virt(nsec_entry, MEM_AREA_IO_SEC);
```

Comment:
@etienne-lms, in plat-ti nsec_entry is not the same thing as in other platforms.